src/lib/viewers/text/CSVViewer.js

```diff
@@ -65,10 +65,14 @@ class CSVViewer extends TextBaseViewer {
                             this.handleDownloadError(error, urlWithAuth);
                         },
                         complete: (results) => {
-                            if (this.isDestroyed() || !results) {
+                            this.data = getProp(results, 'data', []);
+                            if (this.isDestroyed() || this.data.length === 0) {
+                                // If the viewer is destroyed or if we are not able to extract any csv data, error
+                                const error = new PreviewError(ERROR_CODE.LOAD_CSV, __('error_refresh'));
```

Comment:
Should this be a generic error or can the user recover from this state by refreshing?